Documentation/coding-guidelines/project-guidelines.md

```diff
@@ -0,0 +1,116 @@
+```
+## Project .builds file
+To drive the Project build pass we have a *.builds project that will multiplex the various optional build parameters we have for all the various configurtions within a given build pass.
+
+Below is an example for System.Linq.Expressions where it only builds for the windows full build passes and needs a unique build for various target frameworks.
+```
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <ItemGroup Condition="'$(TargetsWindows)' == 'true'">
+    <Project Include="System.Linq.Expressions.csproj">
+      <AdditionalProperties>TargetGroup=</AdditionalProperties> 
+    </Project>
+    <Project Include="System.Linq.Expressions.csproj">
+      <AdditionalProperties>TargetGroup=netcore50aot</AdditionalProperties>
+    </Project>
+    <Project Include="System.Linq.Expressions.csproj">
+      <AdditionalProperties>TargetGroup=netcore50</AdditionalProperties> 
+    </Project>
+    <Project Include="Facade\System.Linq.Expressions.csproj">
+      <AdditionalProperties>TargetGroup=net46</AdditionalProperties>
+    </Project>
+  </ItemGroup>
+  <ItemGroup>
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.traversal.targets))\dir.traversal.targets" />
+</Project>
+```
+##Facades Projects
+Facade projects are unique in that they don't have any code and instead are generated by finding a contract reference assembly with the matching identity and generating type forwards for all the types to where they live in the implementation assemblies (aka facade seeds). There are also partial facades which contain some type forwards as well as some code definitions. Ideally all the various build configurations would be contained in the one csproj file per library but given the unique nature of facades and the fact they are usually a complete fork of everything in the project file it is recommended to create a matching csproj under a Facade directory (<library>\src\Facade\<library>.csproj) and reference that from your .builds file, as in the System.Linq.Expressions example.
+
+TODO: Fill in more information about the required properties for creatng a facade project.
+
+##Defines for code #ifdefs
+Configuration base defines should match the name of the configurtion that is needs the fork in code. So either $(Platform), $(OSGroup), $(ConfigurationGroup), $(TargetGroup). These should only be used as a last resort and we should prefer using partial classes or configuration specific cs files for any necessary forking (see [File naming convention](file-naming-convention)).
+
+Feature based defines should be named like `FEATURE_[FeatureName]`. These can unique to a given library project or potentially shared (via name) across multiple projects.
+
```

Comment:
We need an example, even if it is discouraged.  Otherwise we'll get inconsistent results.